ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Impl/VolDB.hs

```diff
@@ -470,8 +474,14 @@ blockFileParser VolDbArgs{..} = VolDB.Parser $
        fmap (fmap (fmap fst)) -- Drop the offset of the error
      . Util.CBOR.readIncrementalOffsets volHasFS decoder'
   where
-    decoder' :: forall s. Decoder s (Lazy.ByteString -> VolDB.BlockInfo (HeaderHash blk))
-    decoder' = (extractInfo .) <$> volDecodeBlock
+    -- TODO: It looks weird that we use an encoding function 'volEncodeBlock'
+    -- during parsing, but this comes without many performance penalty,
+    -- since the encoding is already cached. We should consider improving
+    -- this, so that it does not create confusion.
+    decoder' :: forall s. Decoder s (Lazy.ByteString
```

Comment:
BTW, I just measured the overhead of this extra encoding. Just removing the bang on the `binaryBlob` field ought to be enough. I then let the ImmutableDB validate 112 epochs from https://github.com/input-output-hk/cardano-mainnet-mirror. I did this 5 times with the bang and 5 times without, I then took the best times of the 5 runs.

With the bang, it took 62.26s.
Without the bang, it took 61.96s.

So there is not noticeable difference between the two, so I see no reason to change this.